src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EmptyStatementCheck.java

```diff
@@ -34,6 +34,24 @@
  *   doConditionalStuff();
  * doUnconditionalStuff();
  * </pre>
+ * <p>Example:</p>
+ * <div class="wrapper">
+ * <pre>
+ * class Expression {
+ *     List&lt;Expression&gt; arguments;
+ *     public void calculate() {
+ *         // Calculates something
+ *     }
+ *
+ *     public bool isConstant() {
+ *         // Returns true, if Expression is constant, else returns false
+ *     }
+ * }
+ * for (int i = 0; i &lt; arguments.size(); arguments.get(i++).calculate()); // OK
+ * while (true); // violation
+ * if (arguments.get(0).isConstant()); // violation
```

Comment:
`else;` would be nice.

Also, code example is not compiliable anyway, you can remove class above.
examples can use some standard things like 
`arguments.get(0).isConstant()` -> `Math.random() > 0.5`
or
`arguments.size(); arguments.get(i++).calculate()` -> `i < 10; i++`

Reply:
> `else;` would be nice.
> 
You mean that I can leave `else; // OK` as an example?
> Also, code example is not compiliable anyway, you can remove class above.
> examples can use some standard things like
> `arguments.get(0).isConstant()` -> `Math.random() > 0.5`
I do not see logic behind `if (Math.random() > 0.5);`, for example.
> or
> `arguments.size(); arguments.get(i++).calculate()` -> `i < 10; i++`